src/nouveau/src/nouveau_fabric_search.erl

```diff
@@ -215,11 +222,43 @@ convert_item(Item) ->
             maps:get(<<"value">>, Item)
     end.
 
-merge_facets(FacetsA, null, _Limit) ->
+top_n_facets(SearchResults, TopN) ->
+    After = SearchResults#{
+        <<"counts">> => top_n_facets(<<"counts">>, SearchResults, TopN),
+        <<"ranges">> => top_n_facets(<<"ranges">>, SearchResults, TopN)
+    },
+    After.
+
+top_n_facets(Key, SearchResults, TopN) ->
+    Facets = maps:get(Key, SearchResults, null),
+    top_n_facets_int(Facets, TopN).
+
+top_n_facets_int(null, _TopN) ->
+    null;
+top_n_facets_int(Facets, TopN) when is_map(Facets), is_integer(TopN), TopN > 0 ->
```

Comment:
`is_map(Facets)` can be replaced with `top_n_facets_int(#{} = Facets,...)` but either way is fine, whichever looks better to you